### PR TITLE
Fully repairable vending machines 

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Revert #24603 
Added the repair based on the broken state of the vending machine to allow fixing it only ones its broken.

## Why / Balance
Fixes https://github.com/space-wizards/space-station-14/issues/24580

## Technical details
C#
When the machine move into broken visuals we add the ``WeldableComponent`` to it.
Then we add the ``WeldableChangedEvent`` event and make the machine go back to full health and working state.
Finally we remove the ``WeldableComponent`` so the machine can only be fixed when its broken.

## Media
- [X] This PR does not require an ingame showcase

## Breaking changes
N/A

**Changelog**
Not going to add one just so it wont show up 2 times?

### CODE REVIEW

**Comment by dvir001** [comment]
Hold up let me make it slower to fix, right now its too fast..

**Comment by dvir001** [comment]
Fixed

**Comment by brainfood1183** [comment]
I think issue with fixing vending machines was that it would refill the vending machine so could be exploited by breaking and fixing to refill. I assume you fixed this issue as well?

**Comment by dvir001** [comment]
> I think issue with fixing vending machines was that it would refill the vending machine so could be exploited by breaking and fixing to refill. I assume you fixed this issue as well?

This is a visual state, but ill test it again to confirm.

**Comment by deltanedas** [comment]
99% sure breaking vending machines is just changing appearance + making them unusable based on damage threshold

File: Resources/Prototypes/Entities/Structures/Machines/vending_machines.yml [hunk 1]
```diff
@@ -53,8 +53,6 @@
       - !type:PlaySoundBehavior
         sound:
           collection: MetalGlassBreak
-  - type: Repairable
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Structures/Machines/vending_machines.yml:56]: i think this should still use Repairable but implement a minDamage field that only starts the doafter if theres at least that much GetTotal() damage

**Comment by EmoGarbage404** [comment]
this is just implementing repairable but in a super strange and roundabout way. Weldable should not be used in this case. As per deltanedas's suggestion, just implement a minimum damage field if it's an issue to be able to repair only slightly broken vendors.

**Comment by potato1234x** [comment]
just make them machines :trollface:

**Comment by dvir001** [comment]
Alright ill do the fix and update when its done, not going to be a hard one to change it seems

**Comment by metalgearsloth** [comment]
Re-open if you come back to it.

**Pull request closed without merging**